### PR TITLE
Workflow file updates

### PR DESCRIPTION
**Pull request by vikasmb**
**What type of PR is this?**
cleanup

**Which issue does this PR fix**:
- Updated workflow files to use go 1.16 for consistency.
- Changed integ test script to use git commit id and deleted unused code
- Cleans stale docker images on runner hosts

**What does this PR do / Why do we need it**:
- Updated workflow files to use go 1.16 for consistency.
 - Changed integ test script to use git commit id and deleted unused code
- Cleans stale docker images on runner hosts
**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No

**Does this change require updates to the CNI daemonset config files to work?**:

No
**Does this PR introduce any user-facing change?**:

No
```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by vikasmb** [comment]
make check-format is failing in unit test workflow. Looking into reason for the same.

File: .github/workflows/build-multi-arch.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Build x64 and ARM64 images 
+name: Build x86 and ARM64 images
```
Comment (haouc) [inline .github/workflows/build-multi-arch.yml:1]: Why do we build x86?

Comment (vikasmb) [reply .github/workflows/build-multi-arch.yml:1]: I believe existing workflow is set to build for x86 mainly for consistency purpose. In this case, build on x86 is also done in unit-test workflow.

**Review by haouc — APPROVED** [review]
lgtm, thanks

**Pull request merged**